src/cls/rbd/cls_rbd_types.cc

```diff
@@ -1124,6 +1129,16 @@ void TrashImageSpec::dump(Formatter *f) const {
   f->dump_unsigned("deferment_end_time", deferment_end_time);
 }
 
+std::ostream& operator<<(std::ostream& os, const TrashImageSpec& spec)
+{
+  return os << "["
+	    << "source=" << spec.source << ", "
+	    << "name=" << spec.name << ", "
+	    << "deletion_time=" << spec.deletion_time << ", "
+	    << "deferment_end_time=" << spec.deferment_end_time
+	    << "]";
+}
```

Comment:
This seems unrelated, why did it become needed?

Reply:
lemme try.

Reply:
yup, it compiles w/o this `operator<<()`.